lib/Fregot/Builtins/Basics.hs

```diff
@@ -122,6 +123,17 @@ builtin_array_concat = Builtin
     (Ty.arrayOf Ty.any 🡒 Ty.arrayOf Ty.any 🡒 Ty.out Ty.unknown) $ pure $
     \(Cons l (Cons r Nil)) -> return (l <> r :: V.Vector Value)
 
+builtin_array_slice :: Monad m => Builtin m
+builtin_array_slice = Builtin
+    (In (In (In Out)))
+    (Ty.arrayOf Ty.any 🡒  Ty.number 🡒  Ty.number 🡒 Ty.out Ty.unknown) $ pure $
```

Comment:
Nitpick: `Ty.out (Ty.arrayOf Ty.unknown)` is a little more precise as return type

Reply:
Oh, of course. Thanks.